app/src/main/java/org/eyeseetea/malariacare/data/database/model/QuestionDB.java

```diff
@@ -1152,6 +1152,9 @@ public void saveValuesDDL(OptionDB optionDB, ValueDB valueDB) {
 
     public void saveValuesText(String answer) {
         ValueDB valueDB = getValueBySession();
+        if (valueDB != null) {
```

Comment:
Can you explain a bit more why this piece of code? It looks like we're doing twice the .setValue() call. I guess that it has to do with the issue #1695 that you say it's solved here? If that were the case, please remove this piece of code from this PR and create a separate PR for #1695, but anyway...please answer my question. Thanks!